Mathlib/Tactic/ExtractGoal.lean

```diff
@@ -93,50 +94,75 @@ example : (X : Nat[X]) = X := by
 ```
 -/
 
-namespace Mathlib.Tactic
+namespace Mathlib.Tactic.ExtractGoal
 
-open Lean Elab Tactic
+open Lean Elab Tactic Meta
 
-/--
-`extract_goal` formats the current goal as a stand-alone theorem or definition,
-and `extract_goal name` uses the name `name` instead of an autogenerated one.
+/-- Have `extract_goal` extract the full local context. -/
+syntax star := "*"
+
+/-- Configuration for `extract_goal` for which variables from the context to include. -/
+syntax config := star <|> (colGt ppSpace ident)*
 
-It tries to produce an output that can be copy-pasted and just work,
+/--
+- `extract_goal` formats the current goal as a stand-alone theorem or definition after
+  cleaning up the local context of irrelevant variables.
+  A variable is *revelant* if (1) it occurs in the target type, (2) there is a relevant variable
+  that depends on it, or (3) the type of the variable is a proposition that depends on a
+  relevant variable.
```

Comment:
I'm not entirely sure what (3) means here... Is there a test case that checks for it?

Reply:
This is documenting the current behavior, since it comes from how the `cleanup` function works. Check out the difference between the first two tests. When you provide variables to `extract_goals` it turns off this third relevancy condition, and so `h₀` is no longer included.

Reply:
Ah okay, that makes sense.